### PR TITLE
NFDIV 3259 added a new superuser event to unlink app1 from case.

### PR DESCRIPTION
**Pull request by aramchurn21**
https://tools.hmcts.net/jira/browse/NFDIV-3259

**Before merging a pull request make sure that:**

- [x] tests have been updated / new tests has been added (if needed)
- [x] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [x] this ticket been reviewed by QA
- [x] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        withEnvironmentVariable("ADMIN_UNLINK_APPLICANT_1_ENABLED", "true")
+            .execute(() -> adminUnlinkApplicant1FromCase.configure(configBuilder));
+
+        assertThat(getEventsFrom(configBuilder).values())
+            .extracting(Event::getId)
+            .contains(ADMIN_UNLINK_APPLICANT_1);
+    }
+
+    @Test
+    public void shouldUnlinkApplicant2() {
```
Comment (tomxelliott) [inline src/test/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCaseTest.java:51]: make sure this reflects applicant1

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCase.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        CaseData caseData = details.getData();
+
+
+        log.info("Admin unlinking Applicant1/Creator from case (id: {})",  details.getId());
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCase.java:50]: technically this is only [CREATOR] right now. Do you want to extend to applicant 1 user role as well?

File: src/test/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCaseTest.java (hunk 1 above)
Comment (aramchurn21) [reply src/test/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCaseTest.java]: Updated naming of unit test.

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCase.java (hunk 2 above)
Comment (aramchurn21) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCase.java]: I updated the log message to simply include Creator, I guess we can leave it as CREATOR, as for this ticket as you mentioned we know the user is the CREATOR. If you were to extend to applicant 1 user role, do we then need to add a new user role (APPLICANT_1) in the UserRole.java class? As currently there are CITIZEN, CREATOR, APPLICANT_2 in this class.

**Review by tomxelliott — APPROVED** [review]

**Comment by amoriartyMOJ** [comment]
Closing as this is no longer required.

**Pull request closed without merging**